### ISSUE TITLE
Too long paths with Orchard 1.9.1 and VS2015

### ISSUE THREAD

**Issue by improwise** [OrchardCMS/Orchard#5561]
I've noticed a but in Orchard 1.9.1. Or rather a bug in Windows 8/VS205, that creates file paths that are to long for Windows to handle. 

To reproduce:

1. Download source package for Orchard 1.9.1 and extract. 
2. Open in Visual Studio 2015 and build
2. Try to delete src folder

This won't work as you will receive an error that file path is to long. More exactly it seems to be Module Orchard.DynamicForm and Orchar.Layout where GULB goes crazy creating super long paths like:

D:\temp\src\Orchard.Web\Modules\Orchard.Layouts\node_modules\gulp-autoprefixer\node_modules\gulp-util\node_modules\dateformat\node_modules\meow\node_modules\indent-string\node_modules\repeating\node_modules\is-finite\node_modules\number-is-nan

Which Windows 8 don't seem to be able to handle. 

(I've moved the folder to a local temp so dont mind first part of path)

**Comment by Skrypt** [comment]
I've just had this issue too while trying to publish a website.

**Comment by dcinzona** [comment]
Need to remove the gulp files.
I actually resolved this by removing gulpfile.js and package.json from orchard.azure.medialibrary, orchard.dynamicforms, and orchard.layouts modules and deleting the newly created folders (node_modules)

If you have trouble deleting the folders, use rimraf...
`npm install rimraf -g`
then go to each module directory in command prompt and execute 
`rimraf node_modules`

**Comment by dcinzona** [comment]
FWIW, I also created a separate project at the root of src with gulpfile.js and package.json and added it to the solution, configured to process my theme css and js files.  Works like a charm and doesn't affect deployment

**Comment by Skrypt** [comment]
I did the same but will look if we can automate this rimraf node_modules thing. @DanielStolt would know better than I.

**Comment by dcinzona** [comment]
Don't automate it. Just run it once and don't have npm install those packages

**Comment by Skrypt** [comment]
I get the idea. I mean, @DanielStolt has been working on this lately so, I guess we could find a better solution than removing it all. I saw somewhere that @sebastienros wrote that it was a npm bug that is now fixed. Maybe upgrading npm would do the trick also.

**Comment by Skrypt** [comment]
I did upgrade npm to version 3.1.3 and still get this error when I'm publishing the website.

**Comment by bleroy** [comment]
Where did @sebastienros write that? As far as I know, npm is mitigating the problem by moving common dependencies up to the root of the package, but does not fix it for good.

**Comment by dcinzona** [comment]
For the VS 2015 deployment to work, you have to remove those directories, and because the path is too long, the only way I was really able to clear them out was using `rimraf`

That said, you want to do that with VS 2015 closed, and only after you removed the package.json and gulpfile.js files from the modules affected.

Once that is done, it should work fine.

Creating another project in the root directory (outside of Orchard.Web) is a bit messy, but it solves the deployment issue.  I literally only have a single project there with those two files in it (gulpfile.js and package.json).  Once that project is included in the solution, when VS 2015 opens the solution up, it will download the node modules and run the gulp tasks.  The trick here becomes merging the 3 modules that use it into one single gulpfile.js - but that shouldn't be too complicated.

**Comment by Skrypt** [comment]
@bleroy #5446 
But without reading it I might be wrong ; it's just that both issues are almost the same.

**Comment by bleroy** [comment]
You're right, it's a duplicate, and he said it.

**Comment by bleroy** [comment]
Duplicate of #5446

**Comment by DanielStolt** [comment]
FWIW, I will fix this proper once and for all as part of #5450.
